core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -984,33 +1007,44 @@ class AutoTuner(
     appendRecommendationForMemoryMB("spark.sql.files.maxPartitionBytes", recommended)
   }
 
+  /**
```

Comment:
Moved the existing logic to an internal method so that it can be overridden by the ProfilerTool